### DISCUSSION TITLE
realmName attribute is empty in Email template

### DISCUSSION THREAD

**Discussion by bstasz-bonrepublic** [keycloak/keycloak#27051] [Q&A] [answered]
Hello,

I am trying to put together a custom theme for emails and I encountered an issue. When running the code locally (docker compose) I get the expected result. But running it in our DEV environment results in the **realmName** attribute being empty when the email is generated.

Our realms are called **dev** and **stage** so no special characters. We do not set display names for the realms. I checked the database and the **name** column is properly set.

Keycloak version: 22.0.5

The function where realmName is used:
```
<#function companyUrl companyName realmName>
  <#if realmName?lower_case == "prod">
      <#return "https://${companyName}.company.com">
  <#else>  
      <#return "https://${companyName}.${realmName?lower_case}.company.com">                    
  </#if>
</#function>
```
Result locally for realm myrealm: https://subdomain.myrealm.company.com
Result in DEV environment for realm stage: https://subdomain..company.com

Thank you!

**Comment by bstasz-bonrepublic** [comment, marked as answer]
What happens is that when the **Realm settings** page is saved, the **Display name** value is saved even if it was not changed. This means that the first time the **Realm settings** page is saved the **displayName** record is created in the database with an empty value. The **realm.displayName** is no longer null but an empty string and retrieved by the template provider.

**Reply by melihsunbul** [reply]
Hello, did you able to solve this issue somehow?

**Comment by bstasz-bonrepublic** [comment]
Is this the expected behaviour or can this be considered a bug?
